### PR TITLE
Upgrade Kafka version from 1.1.1 to 2.7.2

### PR DESCRIPTION
**Pull request by hbehboudi**

### CODE REVIEW

File: src/main/java/ir/sahab/kafkarule/KafkaRule.java [hunk 1]
```diff
@@ -175,21 +178,17 @@ public void createTopic(String topicName) {
...
-        try {
-            // We have to pass the serializer class to ZkClient constructor Otherwise createTopic will return
-            // without error. The topic will exist in zookeeper and be returned when listing topics, but Kafka
-            // itself does not create the topic.
-            zkClient = new ZkClient(zkAddress, 30000, 30000, ZKStringSerializer$.MODULE$);
-            zkUtils = new ZkUtils(zkClient, new ZkConnection(zkAddress), JaasUtils.isZkSecurityEnabled());
-            logger.info("Executing create Topic: " + topicName + ", partitions: " + numPartitions
+        try (KafkaZkClient zkClient = KafkaZkClient
+                .apply(zkAddress, JaasUtils.isZkSaslEnabled(), 30000, 30000, MAX_IN_FLIGHT_REQUESTS,
+                        Time.SYSTEM, METRIC_GROUP, METRIC_TYPE, null, new Some<>(new ZKClientConfig()))) {
```
Comment (bozorgzadeh) [inline src/main/java/ir/sahab/kafkarule/KafkaRule.java]:
```suggestion
                        Time.SYSTEM, METRIC_GROUP, METRIC_TYPE, Option.empty(), Option.empty()) {
```

When some parameter is of type Optional (or Option in Scala), you should never pass null. You should say you have no option. Many APIs just throw an exception if you pass a null value for an argument of type _Optional_. And for the second parameter (ZkClientConfig), you want to say 'I have no special configuration'. So just do not fill this parameter too.

File: src/test/java/ir/sahab/kafkarule/KafkaRuleTest.java [hunk 2]
```diff
@@ -47,7 +48,7 @@ public void testKafkaRuleWithSelfManagedZkServer() {
 
         KafkaConsumer<byte[], byte[]> kafkaConsumer = kafkaRuleWithSelfManagedZk.newConsumer();
         kafkaConsumer.subscribe(Collections.singleton(TOPIC_NAME));
-        ConsumerRecords<byte[], byte[]> records = kafkaConsumer.poll(1000);
+        ConsumerRecords<byte[], byte[]> records = kafkaConsumer.poll(Duration.ofMillis(1000));
```
Comment (bozorgzadeh) [inline src/test/java/ir/sahab/kafkarule/KafkaRuleTest.java:51]: There is a sample in README.md. Please also update the signature of time on that sample too.

**Review by bozorgzadeh — CHANGES_REQUESTED** [review]

File: src/main/java/ir/sahab/kafkarule/KafkaRule.java (hunk 1 above)
Comment (hbehboudi) [reply src/main/java/ir/sahab/kafkarule/KafkaRule.java]: Done

File: src/test/java/ir/sahab/kafkarule/KafkaRuleTest.java (hunk 2 above)
Comment (hbehboudi) [reply src/test/java/ir/sahab/kafkarule/KafkaRuleTest.java:51]: Done

**Pull request merged**